src/ui/public/filter_manager/lib/range.js

```diff
@@ -34,12 +34,24 @@ export default function buildRangeFilter(field, params, indexPattern, formattedV
     };
 
     const knownParams = _.pick(params, (val, key) => { return key in operators; });
-    const script = _.map(knownParams, function (val, key) {
-      // painless expects params.[key] while groovy and expression languages expect [key] only.
-      const valuePrefix = field.lang === 'painless' ? 'params.' : '';
-      return '(' + field.script + ')' + operators[key] + valuePrefix + key;
+    let script = _.map(knownParams, function (val, key) {
+      return '(' + field.script + ')' + operators[key] + key;
     }).join(' && ');
 
+    // We must wrap painless scripts in a lambda in case they're more than a simple expression
+    if (field.lang === 'painless') {
+      const comparators = `boolean gt(Supplier s, def v) {return s.get() > v;}
```

Comment:
Technically you don't need a `;` before a `}` in painless ever. Mad lexer hax.

Anyway, maybe only emit the one of these comparators that you need? It doesn't seem worth it to go through the trouble of building these methods if you aren't going to use them.